### PR TITLE
Implemented mute/unmute A/V for Video Conferencing

### LINKED ISSUE #22: RTC Mute/Unmute Buttons
### Goal: Add Real-Time Conferencing to our application.

Work on branch `video-conferencing`.

I've implemented much of the boilerplate code from the Fireship tutorial for you. Your job is to implement this VideoChat component according to the MVP and implement features such as muting and unmuting video/ audio.

Tutorial:

1. https://github.com/fireship-io/webrtc-firebase-demo/blob/main/main.js

2. https://www.youtube.com/watch?v=WmR9IMUD_CY&ab_channel=Fireship

You can find the full API docs here: https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Signaling_and_video_calling

Keep in mind that we are using the modular version of Firebase: https://firebase.google.com/docs/web/modular-upgrade

These might be helpful, but you may have to consult other resources as well:

1. https://stackoverflow.com/questions/34469618/how-to-add-audio-video-mute-unmute-buttons-in-webrtc-video-chat

2. https://stackoverflow.com/questions/21497566/webrtc-video-stream-stop-sharing?rq=1][1]

Optional: 
1. Check out this cool demo: https://webrtc-from-chat.glitch.me/
- [x] Move ticket
- [x] Familiarize yourself with WebRTC, see tutorial
- [x] See the code I've already implemented from the tutorial. Write descriptive comments explaining your understanding.
- [x] Add unmute/ mute buttons for video **and** audio.
- [x] Build component 
- [x] Test components
- [x] Move on Board
- [x] Make PR, adding replication and testing workflow description (how did you test your feature?)

### PR DESCRIPTION
**Pull request by DylanTom**
## Overview

In this PR, we implemented mute and unmute functionality for audio and video in the video chat feature. We followed mid-fi designs and appropriately styled the page to the best of our abilities. 

## Replication
```
cd backend
yarn install
cd ..
cd frontend
yarn install
yarn run dev
```
On `local`,
+ Start webcam
+ Press the "call" button
+ Copy the `call ID` field
+ Send to `remote`

On `remote`,
+ Start webcam
+ Paste into the `call ID` field
+ Press the "answer" button

## Changes Made

In `VideoChat.tsx` we:
+ Provided documentation for template code
+ Added two functions `toggleAudio()` and `toggleVideo()`
+ Added corresponding buttons for added functions
+ Styled the page according to mid fi

Had to add Tailwind configuration to be consistent with the rest of our codebase

Added icons to `frontend/src/Assets`

## Test Coverage

We tested our sprint using both manual and automated testing.

For automated testing, we created several cypress tests to ensure toggling mute and unmute and video on and off worked as intended. To reproduce, we ran `npx cypress open` in `/frontend`. All test cases pass. 

For manual testing, we started a call to verify the audio and video chat feature worked as intended. 

## Next Steps

In following PRs, we expect high-fis to complete the design of the video chat page. We also expect to have a more concrete understanding of the controls a user has with regards to muting (and unmuting) incoming (and outgoing) media. 

## Related PRs or Issues 

Resolves #22 

## Screenshots

image

  <summary>Video Chat page as displayed in mid-fi</summary>

image

<summary>Passing Cypress tests</summary>

IMG_3932

<summary>Working video connection</summary>

[Video Demo](https://drive.google.com/file/d/1mJIW8bsgkrrGj-dHZz1BaCF32le4rwan/view?usp=sharing)

</details>

### CODE REVIEW

File: backend/functions/sms.js
Comment (biojameskim) [inline backend/functions/sms.js:1]: I don't know if this was part of your sprint but some more documentation on what this is doing would be nice.

File: frontend/src/components/VideoChat.tsx
Comment (biojameskim) [inline frontend/src/components/VideoChat.tsx:1]: This is quite a lot of code so great job on the documentation! Styling should be adjusted to match the midfi's though. 

I'm curious if the connection still works with 2 different users as well?

**Review by biojameskim — COMMENTED** [review]
Great job on the documentation for the VideoChat component. This seems quite large in scope as well so I'm glad to see the MVP functionality is working. Styling can be adjusted but that is probably done best after the features are completed.

**Comment by sriramanved** [comment]
Great job!

**Review by swinshu — COMMENTED** [review]
Already merged - looks good!

**Pull request merged**